### PR TITLE
240 cp apr does not accept params tuple as input

### PR DESCRIPTION
**Pull request by jeremy-myers**
Fixes https://github.com/sandialabs/pyttb/issues/236#issue-1897495997 in a way similar to https://github.com/sandialabs/pyttb/pull/238#issue-1898957283. Also fixes a test in `test_cpapr_pqnr()` where an error on bad line search was thrown before the target of the test was evaluated.

----
:books: Documentation preview :books:: https://pyttb--241.org.readthedocs.build/en/241/

### CODE REVIEW

File: tests/test_cp_apr.py [hunk 1]
```diff
@@ -238,12 +250,18 @@ def test_cpapr_pqnr(capsys):
...
     assert np.isclose(M.full().data, ktensorInstance.full().data, rtol=1e-01).all()
 
+    # Assert that params from previous run can be provided as input
+    M, _, output2 = ttb.cp_apr(sptensorInstance, 2, **output["params"])
+    capsys.readouterr()
+    assert output["params"] == output2["params"]
+
     # Edge cases
     # Confirm timeout works
+    np.random.seed(123)
```
Comment (ntjohnson1) [inline tests/test_cp_apr.py:264]: This seed confirms reproducibility for a given machine but not across different machines. Does it make more sense to just dump an explicit initial guess we know doesn't have an error at the first step? Was the bad line search actually making this test irrelevant or just creating noise in the pytest output? Because it looks like we still captured the time limit exceeded which is the goal of the test anyway.

If its just reducing the print out can we just catch/squash the print. If the previous test was fundamentally broken and the tensor is small can we just use an explicit initial value that doesn't hit the line search issue?

File: tests/test_cp_apr.py [hunk 2]
```diff
@@ -238,12 +250,18 @@ def test_cpapr_pqnr(capsys):
     M, _, _ = ttb.cp_apr(sptensorInstance, 2, algorithm="pqnr")
     capsys.readouterr()
     assert np.isclose(M.full().data, ktensorInstance.full().data, rtol=1e-01).all()
-    M, _, _ = ttb.cp_apr(sptensorInstance, 2, algorithm="pqnr", precompinds=False)
+    M, _, output = ttb.cp_apr(sptensorInstance, 2, algorithm="pqnr", precompinds=False)
     capsys.readouterr()
     assert np.isclose(M.full().data, ktensorInstance.full().data, rtol=1e-01).all()
 
+    # Assert that params from previous run can be provided as input
+    M, _, output2 = ttb.cp_apr(sptensorInstance, 2, **output["params"])
```
Comment (ntjohnson1) [inline tests/test_cp_apr.py:258]: IIRC cp_apr is the slowest of our current tests. Globally, can we also set max iters to a low value throughout the newly added tests since we don't care about the result? In general our test suite runs locally pretty quickly, but trying to prevent gradual bloat.

**Review by ntjohnson1 — COMMENTED** [review]
Thanks! Two minor notes but otherwise looks good.

File: tests/test_cp_apr.py (hunk 1 above)
Comment (dmdunla) [reply tests/test_cp_apr.py:264]: Suggestion: using an initial guess of all ones (weights and factor_matrices) is a typical choice for deterministic testing.

File: tests/test_cp_apr.py (hunk 2 above)
Comment (dmdunla) [reply tests/test_cp_apr.py:258]: Suggestion: use 10 for the maximum number of iterations

Comment (dmdunla) [reply tests/test_cp_apr.py:258]: Or at least enough to make it through a single pass over all modes/dimensions + 1. This will make sure to test outer iteration and inneriteration.

File: tests/test_cp_apr.py (hunk 1 above)
Comment (jeremy-myers) [reply tests/test_cp_apr.py]: I added fixtures for two tensor problems with their solutions as well as two initial guess ktensors. Finding the right combo of problems/initial guesses that ensured PQNR threw a bad L-BFGS error made for a somewhat ugly solution (i.e., 2 tensor inputs & 2 initial guesses), but this ensures that functionality is flexed.

File: tests/test_cp_apr.py (hunk 2 above)
Comment (jeremy-myers) [reply tests/test_cp_apr.py]: IMO this probably warrants its own issue, so I'll open a new issue for it.

**Review by dmdunla — APPROVED** [review]
